src/internal/intrinsic.ts

```diff
@@ -0,0 +1,15 @@
+export type UncapitalizeSTS<T extends string> = string extends T
```

Comment:
Unfortunately this file is necessary.

The built-in Uncapitalize/Lowercase/Uppercase/Capitalize types have what IMO is bad behavior

```ts
type CapitalizeSTS1 = Expect<Equal<CapitalizeSTS<'abc'>, 'Abc'>>
type CapitalizeSTS2 = Expect<Equal<CapitalizeSTS<string>, string>>

type bad1 = Expect<Equal<Capitalize<'abc'>, 'Abc'>>
type bad2 = Expect<Equal<Capitalize<string>, Capitalize<string>>>
```

The `bad2` type test showcases the "bad behavior" around the built-in types.

Why doesn't `Capitalize<string>` resolve to `string`?

This unfortunate behavior affects everything downstream (ie: word casing and object keys). We need utilities that work on both string and literal types, so I had to "rewrite" the built-in types in a way that would work correctly for both string and literal types.